web/models.py

```
@@ -25,6 +27,91 @@
+        return counter.value
+
+
+class UserActivity(models.Model):
+    """Model to track recent user activities for live notifications."""
+
+    ACTIVITY_TYPES = [
+        ("enrollment", "Course Enrollment"),
+        ("quiz_completion", "Quiz Completion"),
+        ("course_completion", "Course Completion"),
+        ("challenge_completion", "Challenge Completion"),
+    ]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)